napari/_vispy/vispy_points_layer.py

```diff
@@ -42,7 +42,7 @@ def _on_data_change(self, event=None):
         # reversed to make the most recently added point appear on top
         # and the rows / columns need to be switch for vispys x / y ordering
         if len(self.layer._indices_view) == 0:
-            data = np.zeros((1, self.layer.dims.ndisplay))
+            data = np.zeros((1, self.layer._dims.ndisplay))
```

Comment:
@sofroniewn for me, it's a bit of a bad smell that the vispy class needs to access a private attribute of the layer... Is this something that is supposed to be fixed in future PRs?

Reply:
yeah, i could actually work on eliminating the references in vispy in this PR. i'll give that a go tomorrow

Reply:
@sofroniewn could you have another look at this before merging?